firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionCoordinator.kt

```diff
@@ -31,30 +28,26 @@ import kotlinx.coroutines.tasks.await
  */
 internal class SessionCoordinator(
   private val firebaseInstallations: FirebaseInstallationsApi,
-  context: CoroutineContext,
   private val eventGDTLogger: EventGDTLoggerInterface,
 ) {
-  private val scope = CoroutineScope(context)
-
-  fun attemptLoggingSessionEvent(sessionEvent: SessionEvent) =
-    scope.launch {
-      sessionEvent.sessionData.firebaseInstallationId =
-        try {
-          firebaseInstallations.id.await()
-        } catch (ex: Exception) {
-          Log.e(TAG, "Error getting Firebase Installation ID: ${ex}. Using an empty ID")
-          // Use an empty fid if there is any failure.
-          ""
-        }
-
+  suspend fun attemptLoggingSessionEvent(sessionEvent: SessionEvent) {
+    sessionEvent.sessionData.firebaseInstallationId =
       try {
-        eventGDTLogger.log(sessionEvent)
-
-        Log.i(TAG, "Successfully logged Session Start event: ${sessionEvent.sessionData.sessionId}")
-      } catch (e: RuntimeException) {
-        Log.e(TAG, "Error logging Session Start event to DataTransport: ", e)
+        firebaseInstallations.id.await()
+      } catch (ex: Exception) {
+        Log.e(TAG, "Error getting Firebase Installation ID: ${ex}. Using an empty ID")
+        // Use an empty fid if there is any failure.
+        ""
```

Comment:
Do we want to allow dispatching events with empty `fid`s? Can we stop from such events to go through? Empty FID could lead to invalid data ingested.

Reply:
I am not sure. @samedson and I talked and thought this was better, we even updated iOS to do the same https://github.com/firebase/firebase-ios-sdk/pull/11161 I am ok with dropping the event instead but we'll need to revert that iOS change to keep it consistent.

Reply:
Do you remember the rationale as to why it is fine to dispatch an event with empty FID? Because, backend is going to drop it anyways since this event would be accounted as invalid.